src/GraphQL.SystemTextJson/OperationMessageJsonConverter.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using GraphQL.Transport;
+
+namespace GraphQL.SystemTextJson
+{
+    /// <summary>
+    /// A custom JsonConverter for reading or writing a <see cref="OperationMessage"/> object.
+    /// </summary>
+    public class OperationMessageJsonConverter : JsonConverter<OperationMessage>
+    {
+        private const string TYPE_KEY = "type";
+        private const string ID_KEY = "id";
+        private const string PAYLOAD_KEY = "payload";
+
+        /// <inheritdoc/>
+        public override void Write(Utf8JsonWriter writer, OperationMessage value, JsonSerializerOptions options)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName(TYPE_KEY);
+            writer.WriteStringValue(value.Type);
+            if (value.Id != null)
+            {
+                writer.WritePropertyName(ID_KEY);
+                writer.WriteStringValue(value.Id);
+            }
+            if (value.Payload != null)
+            {
+                writer.WritePropertyName(PAYLOAD_KEY);
+                JsonSerializer.Serialize(writer, value.Payload, options);
+            }
+            writer.WriteEndObject();
+        }
+
+        /// <inheritdoc/>
+        public override OperationMessage Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+                throw new JsonException();
+
+            var request = new OperationMessage();
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                    return request;
+
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                    throw new JsonException();
+
+                string key = reader.GetString();
+
+                //unexpected end of data
+                if (!reader.Read())
+                    throw new JsonException();
+
+                switch (key)
+                {
+                    case TYPE_KEY:
+                        request.Type = reader.GetString();
+                        break;
+                    case ID_KEY:
+                        request.Id = reader.GetString();
+                        break;
+                    case PAYLOAD_KEY:
+                        request.Payload = JsonSerializer.Deserialize<JsonElement?>(ref reader, options);
```

Comment:
`JsonElement`?

Reply:
`JsonElement` is a struct.  `JsonElement` and `JsonElement?` are both valid.  When deserializing `null`, `JsonElement` will box a value that represents null, while `JsonElement?` will simply return null.

Reply:
So I elected to use the latter, so `Payload` is simply `null` if deserialized as such.

Reply:
I ask about resulting type of payload. Why JsonElement? Why not string, or dictionary or whatever?

Reply:
The purpose is to let the user (in this case the server project) deserialize the payload to the proper type after examining the `type` property.  Because of the dynamic nature of an `OperationMessage`, it cannot be determined ahead-of-time what type the payload might be.  Of course typically it is a `GraphQLRequest` for messages received by the server, and typically a `ExecutionResult` when sending responses to the client.  But there are 'ping' and 'pong' packets and potentially other packet types.

Of course we could write all that logic directly into the `JsonConverter`, but that will tie the converter even more closely to a specific implementation (graphql-ws, graphql-transport-ws, or whatever is next).  Changes to the transport would necessitate a change in the converter even for mundane packet types.  I figured that the server project is really responsible for deciphering the packet; I am just trying to generalize the serialization and deserialization functions.

I also considered writing `OperationMessage` as a generic class, with the payload being of type `T`.  But this would only work if all incoming packets had the same type of payload.

The design here is pretty much the same as is used within the server project.  It's just what seems to make the most sense.

Note that when serializing an `OperationMessage`, you would of course just set `Payload` to an instance of whatever it is to be serialized, and it will be properly serialized.